### PR TITLE
feat: initialize JS components automatically (opt-in) - INNO-1680

### PR DESCRIPTION
**Pull request by yhuard**
To test: 
- run `yarn dist:presets`
- open `examples/auto-init/index.html`

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 70d37f3a9b3b6f05cf7da90d8a69b5e4cb9e9086

https://deploy-preview-1227--europa-component-library.netlify.com

File: src/systems/ec/implementations/vanilla/packages/ec-auto-init/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      const ctor = ECL[componentType];
+      if (typeof ctor !== 'function') {
+        throw new TypeError(
+          `(ecl-auto-init) Could not find '${componentType}'`
+        );
+      }
+
+      if (typeof ctor.autoInit !== 'function') {
+        throw new TypeError(
+          `(ecl-auto-init) Could not find autoInit for  '${componentType}'`
```
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-auto-init/index.js]:
```suggestion
          `(ecl-auto-init) Could not find autoInit for '${componentType}'`
```

File: src/systems/eu/implementations/vanilla/packages/eu-auto-init/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      const ctor = ECL[componentType];
+      if (typeof ctor !== 'function') {
+        throw new TypeError(
+          `(ecl-auto-init) Could not find '${componentType}'`
+        );
+      }
+
+      if (typeof ctor.autoInit !== 'function') {
+        throw new TypeError(
+          `(ecl-auto-init) Could not find autoInit for  '${componentType}'`
```
Comment (emeryro) [inline src/systems/eu/implementations/vanilla/packages/eu-auto-init/index.js]:
```suggestion
          `(ecl-auto-init) Could not find autoInit for '${componentType}'`
```

**Review by emeryro — COMMENTED** [review]
Seems good so far.
Still, we need to list or let the user knows the list of availble values for `data-ecl-auto-init` (it may be hard to guess `FileDownload` for instance)

**Review by emeryro — APPROVED** [review]

**Pull request merged**